### ISSUE TITLE
CLI version?

### ISSUE THREAD

**Issue by danseitz** [innocenat/kindle-epub-fix#11]
Thank you for your work on this! Any hope of it being adapted into a script that can be called from a CLI? Would love to be able to run this on a full library/as part of a post-processing routine. In any case, this works so well so thanks for your time on it!

**Comment by innocenat** [comment]
If you are in a place where you can execute CLI script, maybe using calibre might be a better option? I initially built this for people who aren't using calibre. Granted, over time it has accumulate a few feature that doesn't exist in calibre, but calibre EPUB to EPUB conversion should still fixed most of the issues.

I will see what can I do regarding CLI version, but no promise.

**Comment by danseitz** [comment]
Thanks! I appreciate it. Tried to look over your code and I have tried give it a go but I’m more or less learning from scratch and so it’s slow going. Have tried to use Calibre but it doesn’t have the level of automation I’d like (I’m using the Calibre web server which interestingly doesn’t allow for the automatic conversion of EPUB to EPUB, and there’s no way to schedule a Calibre plugin like Modify ePub to run, has to be done manually. If I could run something like this on my whole library on a schedule (check for fixable errors, skip if none present, fix if present), that would be incredible.

**Comment by innocenat** [comment]
I believe there's Action Chain plugin that allow you to do thing triggered by other action. Never used it so not sure if it can be done what you are trying to do.

**Comment by danseitz** [comment]
Thanks! I’d looked at this awhile back but at the time it didn’t allow for unattended book conversion, but it looks like someone developed a module to accomplish this so I think that will fit the bill!

**Comment by danseitz** [comment]
In case anyone returns with a similar issue, I installed Action Chains for Calibre (use the download plugins feature within the program), then added [this custom event handler code](https://www.mobileread.com/forums/showpost.php?p=4150280&postcount=16). Then I chained that custom event to the built-in action of “convert books” to convert epub to epub. When I add books to Calibre via the web interface, the plugin detects this and runs the convert book function.
